modules/andes-core/client/src/main/java/org/wso2/andes/jndi/PropertiesFileInitialContextFactory.java

```diff
@@ -180,6 +184,46 @@ protected void createConnectionFactories(Map data, Hashtable environment) throws
         }
     }
 
+    /**
+     * Resolve carbon secure vault encrypted properties.
+     * @param environment property values which need to construct the InitialContext
+     */
+    private static void resolveEncryptedProperties(Hashtable environment) {
+
+        if (environment != null) {
+
+            Map<String, String> envVariables = (Hashtable<String, String>) environment;
+            Properties properties = convertToProperties(envVariables);
+            SecretResolver secretResolver = SecretResolverFactory.create(properties);
+
+            for (Object key : envVariables.keySet()) {
+
+                if (secretResolver != null && secretResolver.isInitialized()) {
+                    if (secretResolver.isTokenProtected(key.toString())) {
+                        environment.put(key.toString(), secretResolver.resolve(key.toString()));
+                    }
+                }
+
+            }
+
+        }
+    }
+
+    /**
+     * Convert Map to Properties object.
+     * @param map key value pair details
+     */
+    private static Properties convertToProperties(Map map) {
+
+        Properties prop = new Properties();
+
+        for (Object key : map.keySet()) {
```

Comment:
question, why not iterate entry list?
and I see usage of this method has the knowledge of map being <String>,<String> and here you are not using that knowledge. any reason?